### PR TITLE
i#2626 fp/simd encode: Add support for vector and scalar FMUL.

### PR DESCRIPTION
**Pull request by fhahn**
This uses the encoding scheme introduced in #2811. It also adds INSTR_CREATE_ macros and related tests.

Everything expect the tests are auto-generated now.

Issue: #2626 

Change-Id: I0cb18af8aabf121417404e83d262745e1bbd0a8f

### CODE REVIEW

File: core/arch/aarch64/instr_create.h [hunk 1]
```diff
@@ -47,18 +47,33 @@
  */
 #define FSZ_HALF 1
 
+/**
+ * Operand indication half-precision floating point vector elements.
```
Comment (derekbruening) [inline core/arch/aarch64/instr_create.h]: s/indication/indicating/

Maybe also:
s/elements./elements for the other operands of the containing instruction./

File: core/arch/aarch64/instr_create.h [hunk 2]
```diff
@@ -47,18 +47,33 @@
...
+#define OPND_CREATE_HALF() OPND_CREATE_INT8(FSZ_HALF)
+
 /**
  * Used in an additional immediate source operand to a vector operation, denotes
  * single-precision floating point vector elements. See \ref sec_IR_AArch64.
  */
 #define FSZ_SINGLE 2
 
+/**
+ * Operand indication single-precision floating point vector elements.
```
Comment (derekbruening) [inline core/arch/aarch64/instr_create.h]: Ditto

File: core/arch/aarch64/instr_create.h [hunk 3]
```diff
@@ -47,18 +47,33 @@
...
+#define OPND_CREATE_SINGLE() OPND_CREATE_INT8(FSZ_SINGLE)
+
 /**
  * Used in an additional immediate source operand to a vector operation, denotes
  * double-precision floating point vector elements. See \ref sec_IR_AArch64.
  */
 #define FSZ_DOUBLE 3
 
+/**
+ * Operand indication double-precision floating point vector elements.
```
Comment (derekbruening) [inline core/arch/aarch64/instr_create.h]: Ditto

File: core/arch/aarch64/instr_create.h [hunk 4]
```diff
@@ -557,6 +572,27 @@
...
   INSTR_CREATE_sub_shift(dc, rd, rn, rm_or_imm, sht, sha)
 
+
+/**
+ * Creates a FMUL vector instruction.
+ * \param dc The void * dcontext used to allocate memory for the instr_t.
+ * \param Rd output register
+ * \param Rm input register
+ * \param Rn input register
+ * \param width vector element width as immediate
```
Comment (derekbruening) [inline core/arch/aarch64/instr_create.h]: nit: All 4 of these are nicer as complete sentences: "The output register.", etc.

Maybe for the width you could reference OPND_CREATE_DOUBLE() + the other 2 (just trying to imagine what would make it easy for users).

**Comment by fhahn** [comment]
Done, thanks @derekbruening !

**Pull request merged**